Code/GraphMol/resMolSupplierTest.cpp

```diff
@@ -960,6 +961,19 @@ void testGitHub3349() {
   delete mol;
 }
 
+void testGitHub5406() {
+    BOOST_LOG(rdInfoLog) << "-----------------------\n"
+                         << "testGitHub5406" << std::endl;
+    RWMol *mol;
+    mol = SmilesToMol("CC=[N+]=[N-]");
+    MolOps::addHs(*mol);
+    auto *suppl = new ResonanceMolSupplier((ROMol &)*mol);
+    auto length = suppl->length();
+    TEST_ASSERT(length == 0);
+    delete mol;
+    delete suppl;
```

Comment:
```suggestion
    std::vector<ROMOL_SPTR> mols{"CC=[N+]=[N-]"_smiles, "O=[N+][O-]"_smiles};
    for (auto mol : mols) {
      ResonanceMolSupplier suppl(*mol);
      TEST_ASSERT(suppl.length() == 0);
    }
```

Reply:
Small suggested change to make the test a bit simpler and also add a further test case from #4884, which is indeed caused by the same underlying issue.

Reply:
@bp-kelley : do you plan to add these changes too? I think it'd help to have the tests for #4884 there too.